R/between.R

```diff
@@ -1,25 +1,35 @@
 # is x[i] in between lower[i] and upper[i] ?
-between <- function(x,lower,upper,incbounds=TRUE) {
+between <- function(x,lower,upper,incbounds=TRUE,verbose=getOption("datatable.verbose")) {
   if (is.logical(x)) stop("between has been x of type logical")
   if (is.logical(lower)) lower = as.integer(lower)   # typically NA (which is logical type)
   if (is.logical(upper)) upper = as.integer(upper)   # typically NA (which is logical type)
   is_strictly_numeric <- function(x) is.numeric(x) && !inherits(x, "integer64")
+  if (inherits(x, "POSIXct") && inherits(lower, "POSIXct") && inherits(upper, "POSIXct")) { #3519
```

Comment:
can we add logic for `!is.na(as.POSIXct(lower))` as well?

I do worry about timezones silently mismatching...

Reply:
Only if `x` is already POSIXct, then yes, make sense. But as you noted the tz might be an issue.

Reply:
Yep. How about `if (inherits(x, 'POSIXct') && !is.null(attr(x, 'tzone')))`, we pass the existing `tzone` to `lower`/`upper`.

Reply:
@MichaelChirico sounds fine, you can push to this branch :)

Reply:
Pushed a commit